### PR TITLE
simplelink: use 'errno' instead of '*__errno()' for picolibc compat

### PR DESCRIPTION
**Pull request by keith-packard**
Picolibc doesn't always use an __errno function, preferring to use TLS errno variable instead as that's more efficient. But, that means applications shouldn't dig through the original errno abstraction to use the underlying implementation directly.

Signed-off-by: Keith Packard <keithp@keithp.com>

### CODE REVIEW

**Comment by keith-packard** [comment]
> @keith-packard - I know it's a bit of a hoop, but we need to do this little DNM PR dance for HAL modules. Have you done these before? If not, I'd be happy to walk you through the process.
> 
> https://docs.zephyrproject.org/latest/develop/modules.html#process-for-submitting-changes-to-existing-modules

nope; I've had luck just waiting for patches in the modules to land then submitting a PR with the west.yml change. I've got a handful of module patches pending; will see about getting zephyr PRs submitted.

**Review by cfriedt — APPROVED** [review]
LGTM but will wait for the PR from `zephyrproject-rtos/zephyr` to exercise CI before merging

**Review by stephanosio — APPROVED** [review]

**Comment by keith-packard** [comment]
> LGTM but will wait for the PR from `zephyrproject-rtos/zephyr` to exercise CI before merging

I think I've got all of my module fixes posted as zephyr PRs now.

**Pull request merged**
